addons/website_sale_stock/models/product_template.py

```diff
@@ -17,6 +17,16 @@ class ProductTemplate(models.Model):
     custom_message = fields.Text(string='Custom Message', default='', translate=True)
 
     def _get_combination_info(self, combination=False, product_id=False, add_qty=1, pricelist=False, parent_combination=False, only_template=False):
+        product, __ = self._get_product_combination_info(
+            combination, product_id, parent_combination, only_template)
+
+        virtual_available = 0
+        if product:
+            website = self.env['website'].get_current_website()
+            product = product.sudo().with_context(warehouse=website.warehouse_id.id)
```

Comment:
This should use the same context as below: 
```suggestion
            product = product.sudo().with_context(warehouse=website._get_warehouse_available())
```